### DISCUSSION TITLE
Changing directory with universal arguments with shadow text gives os error

### DISCUSSION THREAD

**Discussion by aadi58002** [minad/consult#1064] [Q&A]
[consult-directory-change-bug.webm](https://github.com/user-attachments/assets/52c2dea5-8391-48a3-a9f0-3c7c3f50f82c)

When changing target path with C-u there seems to be no files list shown at the target location.

Also if the path is rewritten with shadow text. It seems like the command is still given the shadow text path prepended with the correct path (just a guess).

PS: Thanks for the great package

**Comment by minad** [comment]
Hi, please provide a detailed recipe to reproduce the problem. See https://github.com/minad/consult?tab=readme-ov-file#bug-reports.

**Comment by aadi58002** [comment]
The bug is present with minimal config as well.

```emacs-lisp
(add-to-list 'load-path "/home/aditya-yadav/.config/emacs/elpaca/builds/compat")
(add-to-list 'load-path "/home/aditya-yadav/.config/emacs/elpaca/builds/consult")
(add-to-list 'load-path "/home/aditya-yadav/.config/emacs/elpaca/builds/vertico")
(package-initialize)
(require 'consult)
(require 'vertico)
(vertico-mode)
(setq completion-styles '(substring basic))
```

(Actually not sure where to report this bug)
During the universal argument window there seems to be no directory or file list in the completion.
![image](https://github.com/user-attachments/assets/0c43f9dc-c91f-4b60-aa21-e91c15320d9b)

After giving the a new path to universal argument with old path as shadow argument. Ripgrep report back with os error 2.

![image](https://github.com/user-attachments/assets/8b8541ad-715d-44cb-81ce-0c4220f9f9ce)

**Reply by minad** [reply]
So the issue occurs only with shadowed paths?

**Reply by aadi58002** [reply]
Seems like it. I haven't encountered the above error in other day to day use.

**Reply by aadi58002** [reply]
`~/tes/~/tes` was the input and the starting tes was faded (shadow text)

**Reply by minad** [reply]
Thanks. Should be fixed by https://github.com/minad/consult/commit/3d0fc6a8c6a74b21ca854b8632b3f58e0e513b85. I'd appreciate if you can confirm this.

**Reply by aadi58002** [reply]
Yes It works as intended now. Thanks for the fix and for maintaining a great package

**Reply by minad** [reply]
I had to revert the fix. It turns out that the `read-file-name-internal` completion table doesn't work with `completing-read-multiple`, leading to errors. I remember this now - this was the reason why I didn't use `read-file-name-internal` in the first place. This means, the problem here is wontfix. However you can use `vertico-directory-tidy` as `rfn-eshadow-update-overlay` hook.
